File: Cola_Arreglo/Cola.cs
using System;

class Cola {
    const int MAXSIZE = 5;
    static int[] queue = new int[MAXSIZE];
    static int front = -1, rear = -1;

    static void Insertar() {
        if (rear == MAXSIZE - 1) {
            Console.WriteLine("\nDESBORDAMIENTO (OVERFLOW)");
            return;
        }

        Console.Write("\nIngrese el elemento: ");
        int elem = int.Parse(Console.ReadLine());

        if (front == -1 && rear == -1)
            front = rear = 0;
        else
            rear++;

        queue[rear] = elem;
        Console.WriteLine("Elemento insertado correctamente.");
    }

    static void Eliminar() {
        if (front == -1 || front > rear) {
            Console.WriteLine("\nSUBDESBORDAMIENTO (UNDERFLOW)");
            return;
        }

        int elem = queue[front];
        if (front == rear)
            front = rear = -1;
        else
            front++;

        Console.WriteLine("\nElemento eliminado: " + elem);
    }

    static void Mostrar() {
        if (front == -1 || front > rear) {
            Console.WriteLine("\nLa cola está vacía.");
            return;
        }

        Console.WriteLine("\nElementos en la cola:");
        for (int i = front; i <= rear; i++)
            Console.WriteLine(queue[i]);
    }

    static void Main() {
        int opcion;
        do {
            Console.WriteLine("\n*************** MENÚ PRINCIPAL ***************");
            Console.WriteLine("1. Insertar un elemento");
            Console.WriteLine("2. Eliminar un elemento");
            Console.WriteLine("3. Mostrar la cola");
            Console.WriteLine("4. Salir");
            Console.Write("Ingrese su opción: ");
            opcion = int.Parse(Console.ReadLine());

            switch(opcion) {
                case 1: Insertar(); break;
                case 2: Eliminar(); break;
                case 3: Mostrar(); break;
                case 4: Console.WriteLine("\nSaliendo del programa..."); break;
                default: Console.WriteLine("\nOpción inválida. Intente nuevamente."); break;
            }
        } while (opcion != 4);
    }
}

File: Cola_Arreglo/Cola.py
MAXSIZE = 5
queue = [None] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear
    if rear == MAXSIZE - 1:
        print("\nDESBORDAMIENTO (OVERFLOW)")
        return
    
    elem = int(input("\nIngrese el elemento: "))
    
    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1
    
    queue[rear] = elem
    print("Elemento insertado correctamente.")

def eliminar():
    global front, rear
    if front == -1 or front > rear:
        print("\nSUBDESBORDAMIENTO (UNDERFLOW)")
        return
    
    elem = queue[front]
    if front == rear:
        front = rear = -1
    else:
        front += 1
    
    print("\nElemento eliminado:", elem)

def mostrar():
    if front == -1 or front > rear:
        print("\nLa cola está vacía.")
        return
    
    print("\nElementos en la cola:")
    for i in range(front, rear + 1):
        print(queue[i])

# -------- Menú ---------
op = 0
while op != 4:
    print("\n*************** MENÚ PRINCIPAL ***************")
    print("1. Insertar un elemento")
    print("2. Eliminar un elemento")
    print("3. Mostrar la cola")
    print("4. Salir")
    op = int(input("Ingrese su opción: "))

    if op == 1: insertar()
    elif op == 2: eliminar()
    elif op == 3: mostrar()
    elif op == 4: print("\nSaliendo del programa...")
    else: print("\nOpción inválida. Intente nuevamente.")
